### PR TITLE
Saves information of the simulation state after each step.

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

## Summary
Saves information of the simulation state after each step.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

### CODE REVIEW

File: maliput_sim/core/components.py [hunk 1]
```diff
@@ -77,6 +80,14 @@ def update(self, delta_time, sim_state, entity, ecm):
         for component in self.components.values():
             component.update(delta_time, sim_state, entity, ecm)
 
+    def get_state(self):
+        state = {}
+        for component_type, component_list in self.components.items():
+            state[component_type.__name__] = []
+            for component in component_list:
+                state[component_type.__name__].append(component.get_state())
```
Comment (agalbachicar) [inline maliput_sim/core/components.py:88]:
```suggestion
            state[component_type.__name__] = [c.get_state() for c in component_list]
```

File: maliput_sim/core/components.py [hunk 2]
```diff
@@ -104,6 +121,9 @@ def __init__(self, position, rotation):
         self.position = position
         self.rotation = rotation
 
+    def get_state(self):
+        return self.position + self.rotation
+
 
 class Velocity(Component):
```
Comment (agalbachicar) [inline maliput_sim/core/components.py:128]: No state here?

File: maliput_sim/core/components.py [hunk 3]
```diff
@@ -120,3 +140,8 @@ class RoadNetwork(Component):
     def __init__(self, road_network):
         super().__init__()
         self.road_network = road_network
+
+    def get_state(self):
+        # TODO: Evaluate if this is the best way to store the maliput road network. RoadNetwork is stateless
+        #       and probably doesn't need to be stored in the ECM.
```
Comment (agalbachicar) [inline maliput_sim/core/components.py:146]:
```suggestion
        # TODO: Evaluate if this is the best way to store the maliput road network.
```

It is not stateless (think of the controllers for the phases, state providers in general). However, I see the point of not returning anything.

File: maliput_sim/simulation.py [hunk 4]
```diff
@@ -96,19 +108,39 @@ def step_for(self, duration):
...
+
+        Returns:
+            The road network.
+        """
+        rn_entities = self._entity_component_manager.get_entities_with_component(RoadNetwork)
+        if rn_entities is None:
+            return None
+        road_network_component = rn_entities[0].get_components(RoadNetwork)[0]
+
+        return road_network_component.road_network
```
Comment (agalbachicar) [inline maliput_sim/simulation.py:146]:
```suggestion
        return rn_entities[0].get_components(RoadNetwork)[0].road_network
```

**Review by agalbachicar — CHANGES_REQUESTED** [review]
PTAL

File: maliput_sim/core/components.py (hunk 2 above)
Comment (francocipollone) [reply maliput_sim/core/components.py:126]: Nice catch!

File: maliput_sim/simulation.py (hunk 4 above)
Comment (francocipollone) [reply maliput_sim/simulation.py:146]: Done

File: maliput_sim/core/components.py (hunk 3 above)
Comment (francocipollone) [reply maliput_sim/core/components.py]: Good point

**Review by agalbachicar — APPROVED** [review]
LGTM

**Pull request merged**